### DISCUSSION TITLE
"Unable to locate secure storage module" when building with Eclipse first

### DISCUSSION THREAD

**Discussion by Bananeweizen** [eclipse-tycho/tycho#2645] [Q&A] [answered]
We use Artifactory repositories with authentication in our target platform. We've meanwhile recognized a confusing error pattern:
* If we set up a new Eclipse IDE, and build from that IDE, the target resolution of PDE will query the credentials and the PDE build is fine with those credentials. However, every Maven build will afterwards fail with `org.eclipse.equinox.security.storage.StorageException: Unable to locate secure storage module` when trying to access the secure store.
* If we delete the secure store, and run a Maven build without opening Eclipse in between, the secure store will be created during the build without any further querying of credentials. This is because we have server entries with user/password in the settings.xml for every URL in the target platform, so credentials are generally available to Maven. Afterwards we can build both with Eclipse and Maven, using the maven generated secure store.

We do recognize the encrypted entries in the secure store file have different sizes, depending on whether they are created by Eclipse or Maven, and therefore suspect that Eclipse choose an encryption algorithm by default which is not understood by Maven, but that Maven choose an encryption by default which is understood by Eclipse.

Any thoughts on that? Anything that we could check or configure to make this work out of the box also with credentials provided in Eclipse?

**Comment by laeubi** [comment, marked as answer]
Tycho can't use the credentials from Eclipse, you must always configure this with maven.
About the error printout there is/was an issue at equinox but no one cared enough to investigate or fix this:
- https://github.com/eclipse-equinox/equinox/issues/99

in the meanwhile Tycho 4+ has migrated completely away from the "secure" storage and should not use this anywhere directly.
